### PR TITLE
small dual-funding related nits

### PR DESCRIPTION
**Pull request by niftynei**
A bunch of small memory errors/leaks/documentation nits/spacing fixes for dual-funded related code.

### CODE REVIEW

File: lightningd/dual_open_control.c [hunk 1]
```diff
@@ -679,7 +679,9 @@ wallet_commit_channel(struct lightningd *ld,
 		return NULL;
 	}
 
-	channel_info->fee_states = new_fee_states(uc, opener, &feerate);
+	channel_info->fee_states = take(new_fee_states(uc,
+						       opener,
+						       &feerate));
```
Comment (rustyrussell) [inline lightningd/dual_open_control.c:683]: That is almost certainly wrong.  Were you thinking `tal_steal`?  In which case, change `uc` to `channel_info`.

**Review by rustyrussell — CHANGES_REQUESTED** [review]
Just one weirdness needs fixing.

I also flinch when I see 'fixup' since that has a specific git meaning, prehaps 'cleanup'.  Or more commonly, label the part you're changing, eg 'lightningd:'

File: lightningd/dual_open_control.c (hunk 1 above)
Comment (niftynei) [reply lightningd/dual_open_control.c:685]: we call `new_channel` below, which if `taken`, will properly tal_steal it onto the newly created `channel` instance. updated to make this clearer, and set the context to `NULL`.

**Comment by niftynei** [comment]
> I also flinch when I see 'fixup' since that has a specific git meaning, prehaps 'cleanup'. Or more commonly, label the part you're changing, eg 'lightningd:'

Great point, updated the comments to not have 'fixup' in the name.

**Comment by rustyrussell** [comment]
Trivial rebase

**Comment by rustyrussell** [comment]
Trivial whitespace fix...

**Comment by rustyrussell** [comment]
Ack https://github.com/ElementsProject/lightning/pull/4061/commits/d00ad9ea10a45e6ee5df8d26b47e3f2193026e50

**Review by rustyrussell — APPROVED** [review]

**Pull request merged**